File: app/model/db/resume.py
#No actual db for now but it makes sense to have an internal model for the resume which can mimic a potential db schema
from pydantic import BaseModel
from typing import List, Optional

class ResumeEducation(BaseModel):
    degree: Optional[str] = None
    major: Optional[str] = None
    school: Optional[str] = None
    start: Optional[str] = None
    end: Optional[str] = None

class ResumeExperience(BaseModel):
    title: Optional[str] = None
    company: Optional[str] = None
    start: Optional[str] = None
    end: Optional[str] = None
    description: Optional[str] = None

class AISummary(BaseModel):
    quick_summary: Optional[str] = None
    general_employability_rating: Optional[int] = None


class Resume(BaseModel):
    summary: Optional[str] = None
    education: Optional[List[ResumeEducation]] = None
    experience: Optional[List[ResumeExperience]] = None
    skills: Optional[List[str]] = None
    certifications: Optional[List[str]] = None
    languages: Optional[List[str]] = None
    publications: Optional[List[str]] = None

class SummarizedResume(BaseModel):
    resume: Resume
    ai_summary: AISummary

File: frontend/src/types/resume.ts

interface StandardizeResumeOutEducation {
    degree: string | null;
    major: string | null;
    school: string | null;
    start: string | null;
    end: string | null;
}

interface StandardizeResumeOutExperience {
    title?: string | null;
    company?: string | null;
    start: string | null;
    end: string | null;
    description: string | null;
}

export interface StandardizeResumeOut {
    summary?: string | null;
    education?: StandardizeResumeOutEducation[] | null;
    experience?: StandardizeResumeOutExperience[] | null;
    skills?: string[] | null;
    certifications?: string[] | null;
    languages?: string[] | null;
    publications?: string[] | null;
}


export interface SummarizeResumeOut {
    quick_summary: string | null;
    general_employability: number | null;
}